lib/uplinkc/encryption_access.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+// #include "uplink_definitions.h"
+import "C"
+import (
+	"unsafe"
+
+	"storj.io/storj/lib/uplink"
+	"storj.io/storj/pkg/storj"
+)
+
+//export new_encryption_access
+// new_encryption_access creates an encryption access context
+func new_encryption_access(cerr **C.char) C.EncryptionAccessRef {
+	return C.EncryptionAccessRef {
+		_handle:  universe.Add(uplink.NewEncryptionAccess()),
+	}
+}
+
+//export new_encryption_access_with_default_key
+// new_encryption_access creates an encryption access context with a default key set.
+func new_encryption_access_with_default_key(key *C.uint8_t) C.EncryptionAccessRef {
+	goKey, cKey := storj.Key{}, (*storj.Key)(unsafe.Pointer(key))
+	copy(goKey[:], cKey[:])
+
+	return C.EncryptionAccessRef {
+		_handle:  universe.Add(uplink.NewEncryptionAccessWithDefaultKey(goKey)),
+	}
+}
+
+//export set_default_key
+// set_default_key sets the default key for the encryption access context.
+func set_default_key(encAccessRef C.EncryptionAccessRef, key *C.uint8_t, cerr **C.char) {
+	encAccess, ok := universe.Get(encAccessRef._handle).(*uplink.EncryptionAccess)
+	if !ok {
+		*cerr = C.CString("invalid encryption access")
+		return
+	}
+
+	goKey, cKey := storj.Key{}, (*storj.Key)(unsafe.Pointer(key))
+	copy(goKey[:], cKey[:])
```

Comment:
We now use `SaltedKeyFromPassphrase` instead of plain `copy` to convert the human-readable passphrase to an encryption key.

Reply:
usage updated :+1: